src/priv_verif_token.ts

```diff
@@ -0,0 +1,273 @@
+// Copyright (c) 2023 Cloudflare, Inc.
+// Licensed under the Apache-2.0 license found in the LICENSE file or at https://opensource.org/licenses/Apache-2.0
+
+import {
+    Evaluation,
+    EvaluationRequest,
+    FinalizeData,
+    Oprf,
+    VOPRFClient,
+    VOPRFServer,
+    generateKeyPair,
+    type DLEQParams,
+    type Group,
+    type SuiteID,
+    type HashID,
+    DLEQProof,
+} from '@cloudflare/voprf-ts';
+
+import { joinAll } from './util.js';
+import {
+    AuthenticatorInput,
+    Token,
+    TokenChallenge,
+    type TokenTypeEntry,
+} from './auth_scheme/private_token.js';
+
+export interface VOPRFExtraParams {
+    suite: SuiteID;
+    group: Group;
+    Ne: number;
+    Ns: number;
+    Nk: number;
+    hash: HashID;
+    dleqParams: DLEQParams;
+}
+
+const VOPRF_SUITE = Oprf.Suite.P384_SHA384;
+const VOPRF_GROUP = Oprf.getGroup(VOPRF_SUITE);
+const VOPRF_HASH = Oprf.getHash(VOPRF_SUITE) as HashID;
+const VOPRF_EXTRA_PARAMS: VOPRFExtraParams = {
+    suite: VOPRF_SUITE,
+    group: VOPRF_GROUP,
+    Ne: VOPRF_GROUP.eltSize(),
+    Ns: VOPRF_GROUP.scalarSize(),
+    Nk: Oprf.getOprfSize(VOPRF_SUITE),
+    hash: VOPRF_HASH,
+    dleqParams: {
+        gg: VOPRF_GROUP,
+        hashID: VOPRF_HASH,
+        hash: Oprf.Crypto.hash,
+        dst: '',
+    },
+} as const;
+
+// Token Type Entry Update:
+//  - Token Type VOPRF (P-384, SHA-384)
+//
+// https://datatracker.ietf.org/doc/html/draft-ietf-privacypass-protocol-16#name-token-type-voprf-p-384-sha-
+export const VOPRF: Readonly<TokenTypeEntry> & VOPRFExtraParams = {
+    value: 0x0001,
+    name: 'VOPRF (P-384, SHA-384)',
+    Nid: 32,
+    publicVerifiable: false,
+    publicMetadata: false,
+    privateMetadata: false,
+    ...VOPRF_EXTRA_PARAMS,
+} as const;
+
+export function keyGen2(): Promise<{ privateKey: Uint8Array; publicKey: Uint8Array }> {
+    return generateKeyPair(VOPRF.suite);
+}
+
+async function getTokenKeyID(publicKey: Uint8Array): Promise<Uint8Array> {
+    return new Uint8Array(await crypto.subtle.digest('SHA-256', publicKey));
+}
+
+export class TokenRequest2 {
```

Comment:
suffixing with `2` is not ideal, and confusing because this is being defined to token type 0x0001.
Why not simply export it as `TokenRequest` and let downstream users rename it on import?

Reply:
oh yes, this was an outdated version. please take a look on the revised version.